src/http-api.js

```diff
@@ -385,24 +391,45 @@ module.exports.MatrixHttpApi.prototype = {
         if (!queryParams) {
             queryParams = {};
         }
-        if (!queryParams.access_token) {
-            queryParams.access_token = this.opts.accessToken;
+        if (this.useAuthorizationHeader) {
+            if (isFinite(opts)) {
+                // opts used to be localTimeoutMs
```

Comment:
> I have to check if the parameter is an int or not

Why? The docs seem pretty clear that it shouldn't be an int. Either:
 - the docs are wrong. You got errors as a result, in which case I want to know what the error is and see if we can fix the function which is calling this with an int instead of an object.
 - the docs aren't wrong, meaning this is not required.